### PR TITLE
feat: add release history

### PR DESCRIPTION
**Pull request by GlennBullingham**
- adds a Release Notes section to the TOC
 - adds a version history page

Signed-off-by: GlennBullingham <glenn.bullingham@datacore.com>

### CODE REVIEW

File: SUMMARY.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Table of contents
 
-* [Welcome to Mayastor!](README.md)
+* [Welcome to Mayastor!](README.md)]
```
Comment (anupriya0703) [inline SUMMARY.md]: There is an extra ']'.

**Review by anupriya0703 — APPROVED** [review]

File: SUMMARY.md (hunk 1 above)
Comment (GlennBullingham) [reply SUMMARY.md]: Good spot, thank you.  I've corrected it.

**Comment by GlennBullingham** [comment]
In addition to the typo in the TOC file, there were also unsaved changes on my local branch relating to the formatting of the history.md file which I have now pushed, so you may wish to take another look there too.  Thanks.

File: releases/history.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Release History
+
+## Version 1.0
+* ### Lastest Release: [v1.0.3](https://github.com/openebs/mayastor/releases/tag/v1.0.3)
+  * released: 11-Oct-2022
+* ### Patch Releases:
+  * [v1.0.1](https://github.com/openebs/mayastor/releases/tag/v1.0.1)
+  * [v1.0.2](https://github.com/openebs/mayastor/releases/tag/v1.0.2)
```
Comment (avishnu) [inline releases/history.md:7]: Should v1.0.2 precede v1.0.1 ? Latest at the top, followed by others with oldest at the bottom?

Comment (GlennBullingham) [reply releases/history.md:7]: I'll be honest, I was copying from kubernetes.io homework.  They list in a similar fashion, using ascending order.  So I'm "following suit" with something that users will have seen elsewhere.

Comment (GlennBullingham) [reply releases/history.md:7]: However, with time to consider it, I think it is logical to go in reverse chronological order here.  I'll update.

**Review by avishnu — APPROVED** [review]
lgtm

**Pull request merged**